packages/go_router/test/go_router_test.dart

```diff
@@ -4909,6 +4909,54 @@ void main() {
       expect(statefulWidgetKeyB.currentState?.counter, equals(1));
     });
   });
+
+  ///Regression tests for https://github.com/flutter/flutter/issues/132557
+  group('overridePlatformDefaultLocation', () {
+    test('No initial location provided', () {
+      expect(
+          () => GoRouter(
+                overridePlatformDefaultLocation: true,
+                routes: <RouteBase>[
+                  GoRoute(
+                    path: '/a',
+                    builder: (BuildContext context, GoRouterState state) =>
+                        const Placeholder(),
+                  ),
+                  GoRoute(
+                    path: '/b',
+                    builder: (BuildContext context, GoRouterState state) =>
+                        const Placeholder(),
+                  ),
+                ],
+              ),
+          throwsA(const TypeMatcher<AssertionError>()));
+    });
+    test('Test override using routeInformationProvider', () {
+      final String platformRoute =
```

Comment:
Sorry I wasn't thinking clearly, I realize the initialRoute will take priority if WidgetsBinding.instance.platformDispatcher.defaultRouteName is `/`.

Can you convert this test to `testWidget`

and use
```
tester.binding.platformDispatcher.defaultRouteNameTestValue('/some-route')
```
This should mock the platform default initial location. and then verify the `GoRouter(initialLocation:)` really take priority

Reply:
Implemented